tools/.golangci.yaml

```diff
@@ -17,6 +18,47 @@ linters:
     - usetesting
     - whitespace
   settings:
+    depguard:
+      rules:
+        api:
+          list-mode: lax
+          files:
+            - '**/api/**/*.go'
+            - '!**/server/etcdserver/api/**/*.go'
+          deny:
+            - pkg: go.etcd.io/etcd/api/v3$
+            - pkg: go.etcd.io/etcd/pkg/v3
+            - pkg: go.etcd.io/etcd/v3
+            - pkg: go.etcd.io/tests/v3
+            - pkg: go.etcd.io/v3
+        client:
+          list-mode: lax
+          files:
+            - "**/client/v3/**/*.go"
+          deny:
+            - pkg: go.etcd.io/etcd/pkg/v3
+            - pkg: go.etcd.io/etcd/v3
+            - pkg: go.etcd.io/tests/v3
+        cli-and-server:
+          list-mode: lax
+          files:
+            - "**/etcdctl/**/*.go"
+            - "**/etcdutl/**/*.go"
+            - "**/server/**/*.go"
+            - "!**/tests/antithesis/server/**/*.go"
+          deny:
+            - pkg: go.etcd.io/etcd/v3
+            - pkg: go.etcd.io/tests/v3
+        pkg:
+          list-mode: lax
+          files:
+            - "**/pkg/**/*.go"
+            - "!**/client/pkg/**/*.go"
+            - "!**/tools/rw-heatmaps/pkg/**/*.go"
```

Comment:
suggest not to mixes multiple modules.
ditto as https://github.com/etcd-io/etcd/pull/20721#discussion_r2378336321.

Reply:
Same here. This is not mixing multiple modules, it's targeting only the root `pkg`, excluding other `pkg` directories in the repo.